File: 643. Maximum Average Subarray I/643. Maximum Average Subarray I.java
class Solution {

  public double findMaxAverage(int[] nums, int k) {
    double currSum = 0;
    for (int i = 0; i < k; i++) {
      currSum += nums[i];
    }

    double maxAvg = currSum / k;
    for (int i = k; i < nums.length; i++) {
      currSum += nums[i] - nums[i - k];
      maxAvg = Math.max(maxAvg, currSum / k);
    }

    return maxAvg;
  }
}

File: 643. Maximum Average Subarray I/643. Maximum Average Subarray I.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
  let currSum = 0;
  for (let i = 0; i < k; i++) {
    currSum += nums[i];
  }

  let maxAvg = currSum / k;
  for (let i = k; i < nums.length; i++) {
    currSum += nums[i] - nums[i - k];
    maxAvg = Math.max(maxAvg, currSum / k);
  }

  return maxAvg;
};
